python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py

```diff
@@ -21,13 +21,66 @@
 from . import PhoSimDESCQA, PhoSimDESCQA_AGN
 from . import bulgeDESCQAObject_protoDC2 as bulgeDESCQAObject, \
     diskDESCQAObject_protoDC2 as diskDESCQAObject, \
-    knotsDESCQAObject_protoDC2 as knotsDESCQAObject, \
-    agnDESCQAObject_protoDC2 as agnDESCQAObject, \
-    TwinklesCompoundInstanceCatalog_DC2 as twinklesDESCQACompoundObject, \
-    sprinklerCompound_DC2 as sprinklerDESCQACompoundObject, \
-    TwinklesCatalogZPoint_DC2 as DESCQACat_Twinkles
-
-__all__ = ['InstanceCatalogWriter', 'make_instcat_header', 'get_obs_md']
+    knotsDESCQAObject_protoDC2 as knotsDESCQAObject
+from . import DC2PhosimCatalogSN, SNFileDBObject
+
+__all__ = ['InstanceCatalogWriter', 'make_instcat_header', 'get_obs_md',
+           'snphosimcat']
+
+
+SNDTYPESR1p1 = np.dtype([('snid_in', int),
+                         ('x0_in', float),
+                         ('t0_in', float),
+                         ('x1_in', float),
+                         ('c_in', float),
+                         ('z_in', float),
+                         ('snra_in', float),
+                         ('sndec_in', float)])
+
+def snphosimcat(fname, tableName, obs_metadata, objectIDtype, idColKey='snid_in',
+              delimiter=',', dtype=SNDTYPESR1p1):
+    """convenience function for writing out phosim instance catalogs for
+    different SN populations in DC2 Run 1.1 that have been serialized to
+    csv files.
+
+    Parameters: string
+	absolute path to csv file for SN population.
+    tableName : string 
+	table name describing the population upto user choice.
+    obs_metadata: instance of `lsst.sims.utils.ObservationMetaData`
+	observation metadata describing the observation
+    idColKey : string, defaults to values for Run1.1
+        describes the input parameters to the database as interpreted from the
+        csv file.
+    delimiter : string, defaults to ','
+        delimiter used in the csv file
+    dtype : instance of `numpy.dtype`
+	tuples describing the variables and types in the csv files.
+    """
+    
+    x = SNFileDBObject(fname, runtable=tableName,
```

Comment:
I'd feel more comfortable using more explanatory variable names than `x` and `y`.  Maybe `dbobj` and `cat`?